### ISSUE TITLE
AssertionError: backend.__version__ == __version__[:3])

### ISSUE THREAD

**Issue by ardydedase** [couchbaselabs/couchbase-python-cffi#3]
Assigning the issue here as suggested by Burhan Khalid on Stackoverflow: http://stackoverflow.com/questions/32063227/python-couchbase-cffi-assertionerror-backend-version-version-3

Full error text:

```
No module named couchbase._libcouchbase
Traceback (most recent call last):
  File "app_main.py", line 75, in run_toplevel
  File "runtests.py", line 12, in <module>
    mod = __import__(t, globals(), locals(), ['suite'])
  File "/home/travis/build/ardydedase/pycouchbase/tests/test_pycouchbase.py", line 15, in <module>
    from pycouchbase.utils import *
  File "/home/travis/build/ardydedase/pycouchbase/pycouchbase/__init__.py", line 11, in <module>
    import couchbase_ffi as couchase
  File "/home/travis/virtualenv/pypy-2.5.0/site-packages/couchbase_ffi-0.2.0.0-py2.7-linux-x86_64.egg/couchbase_ffi/__init__.py", line 50, in <module>
    import couchbase_ffi._libcouchbase as _libcouchbase
  File "/home/travis/virtualenv/pypy-2.5.0/site-packages/couchbase_ffi-0.2.0.0-py2.7-linux-x86_64.egg/couchbase_ffi/_libcouchbase.py", line 3, in <module>
    from couchbase_ffi._cinit import get_handle
  File "/home/travis/virtualenv/pypy-2.5.0/site-packages/couchbase_ffi-0.2.0.0-py2.7-linux-x86_64.egg/couchbase_ffi/_cinit.py", line 13, in <module>
    ffi = FFI()
  File "/home/travis/virtualenv/pypy-2.5.0/site-packages/cffi/api.py", line 59, in __init__
    backend.__version__ == __version__[:3])
AssertionError
```

This only happens when I try to `import couchbase_ffi` in the PyPy build on Travis: https://travis-ci.org/ardydedase/pycouchbase/jobs/76043402#L1663

Details:

I have this in my `requirements.txt` for `couchbase-python-cffi`:

```
cffi==0.8.0
couchbase==2.0
```

Looks like an incompatibility issue with the packages. But not sure where it's coming from.

**Comment by mnunberg** [comment]
Does this happen still if you use the tested version of cffi (the one we use in our own travis.yml)?

**Comment by ardydedase** [comment]
Yes it still happens, that was my previous setting. It was using cffi `1.1.2` in that setting `cffi>=0.8.0`. I only tried the cffi fixed version in the most recent build just to see whether `0.8.0` would work.

**Comment by mnunberg** [comment]
The pypy version here may also be relevant. I'll try to test this a bit later and see what I come up with.

**Comment by mnunberg** [comment]
It seems that the cffi version is fixed inside pypy and unchangeable. I just tried with pypy 2.6 and cffi 1.1.0 and it still all seems to work here.

**Comment by mnunberg** [comment]
One more thing. I see in your code that you do `import couchbase_ffi as couchbase`. You shouldn't alias `couchbase_ffi` as anything. It's not a _replacement_ of the `couchbase` module but rather an injection of specific routines inside `couchbase._libcouchbase`. Therefore, you only need to import the line, and then use the rest of your imports as usual.. therefore:

``` python
try:
  import couchbase
except ImportError:
  import couchbase_ffi
  import couchbase
```

**Comment by mnunberg** [comment]
Looking at the line at which the error originates, this doesn't really seem to have much to do with the couchbase module or its ffi extensions. There seems to be something actually wrong with your ffi installation there in travis. To test this, simply try to import cffi as you normally would, and instantiate the FFI object directly (via `FFI()`). This should display the same error.

I can't reproduce this on my local environment (OS X; 2.6 pypy, Linux; 2.4 pypy) but it may be a result of a specific vendor's (debian's) decisions because of one reason or another.

I will try to download your actual code and see if I can reproduce it then. Maybe it's a venv issue?

**Comment by ardydedase** [comment]
It's passing the PyPy tests now: https://travis-ci.org/ardydedase/pycouchbase. I forced it to use `cffi==1.1.0` because you mentioned that `1.1.0` worked for you. I also did the fixes you suggested for  `import couchbase_cffi`. Previously I was only using either cffi `0.8.0` or `1.1.2`. So it looks like there's something about `1.1.2` and `0.8.0` that the build is not happy about. Could be incompatible with the PyPy version on Travis.

Thanks a lot for your help!
